rules/sort-interfaces.ts

```diff
@@ -215,11 +238,22 @@ export default createEslintRule<Options<string[]>, MESSAGE_ID>({
                 name,
               }
 
-              if (
-                options.partitionByNewLine &&
-                lastElement &&
-                getLinesBetween(sourceCode, lastElement, elementSortingNode)
-              ) {
+              let shouldPartition = false
+              if (partitionComment) {
+                let comments = getCommentsBefore(element, sourceCode)
+                shouldPartition = hasPartitionComment(
+                  partitionComment,
+                  comments,
+                )
+              } else if (options.partitionByNewLine) {
```

Comment:
I think that this `else if` prevents `partitionByComment` and `partitionByNewLine` from co-existing together, what do you think about

```ts
let lastElement = accumulator.at(-1)?.at(-1)
if (
  (partitionComment &&
    hasPartitionComment(
      partitionComment,
      getCommentsBefore(element, sourceCode),
    )) ||
  (options.partitionByNewLine &&
    lastElement &&
    getLinesBetween(sourceCode, lastElement, sortingNode))
) {
  accumulator.push([])
}
```